cadctl/cmd/investigate/investigate.go

```diff
@@ -0,0 +1,289 @@
+
+	"github.com/spf13/cobra"
+)
+
+// InvestigateCmd represents the entry point for alert investigation
+var InvestigateCmd = &cobra.Command{
+	Use:   "investigate",
+	Short: "Filter for and investigate supported alerts",
+	RunE:  run,
+}
+
+var (
+	payloadPath = "./payload.json"
+)
+
+func init() {
+	const payloadPathFlagName = "payload-path"
+	InvestigateCmd.Flags().StringVarP(&payloadPath, payloadPathFlagName, "p", payloadPath, "the path to the payload")
+}
+
+func run(cmd *cobra.Command, args []string) error {
+
+	fmt.Println("Running CAD with webhook payload:")
+	payload, err := os.ReadFile(payloadPath)
+	if err != nil {
+		return fmt.Errorf("failed to read webhook payload: %w", err)
+	}
+	fmt.Printf("%s\n", string(payload))
+
+	pdClient, err := GetPDClient(payload)
+	if err != nil {
+		return fmt.Errorf("could not start pagerdutyClient: %w", err)
+	}
+
+	// put intro struct in pd for simpler access
+	title := pdClient.GetTitle()
+	serviceName := pdClient.GetServiceName()
+	incidentID := pdClient.GetIncidentID()
+	eventType := pdClient.GetEventType()
+
+	// After this point we know CAD will investigate the alert
+	// so we can initialize the other clients
+	alertType := isAlertSupported(title, serviceName)
+	if alertType == "" {
+		fmt.Printf("Alert is not supported by CAD: %s", title)
+		return nil
+	}
+
+	// print parsed struct
+	fmt.Printf("AlertType is '%s'\n", alertType)
+	fmt.Printf("Incident ID is: %s\n", incidentID)
+
+	externalClusterID, err := pdClient.RetrieveExternalClusterID()
+	if err != nil {
+		return fmt.Errorf("GetExternalID failed on: %w", err)
+	}
+
+	awsClient, err := GetAWSClient()
+	if err != nil {
+		return fmt.Errorf("could not start awsClient: %w", err)
+	}
+
+	ocmClient, err := GetOCMClient()
+	if err != nil {
+		return fmt.Errorf("could not create ocm client: %w", err)
+	}
+
+	// Try to jump into support role
+	customerAwsClient, err := jumpRoles(awsClient, ocmClient, pdClient,
+		externalClusterID, incidentID)
+	if err != nil {
+		return err
+	}
+
+	// Alert specific setup of investigation.Client
+	client := investigation.Client{}
+
+	cluster, err := ocmClient.GetClusterInfo(externalClusterID)
+	if err != nil {
+		return fmt.Errorf("could not retrieve cluster info for %s: %w", externalClusterID, err)
+	}
+
+	clusterdeployment, err := ocmClient.GetClusterDeployment(cluster.ID())
+	if err != nil {
+		return fmt.Errorf("could not retrieve Cluster Deployment for %s: %w", cluster.ID(), err)
+	}
+
+	switch alertType {
+	case "ClusterHasGoneMissing":
+		_, err = checkCloudProvider(&ocmClient, externalClusterID, []string{"aws"})
+		if err != nil {
+			return fmt.Errorf("failed cloud provider check: %w", err)
+		}
+
+		client = investigation.Client{
+			Investigation: &chgm.Client{
+				Service: chgm.Provider{
+					AwsClient: customerAwsClient,
+					OcmClient: ocmClient,
+					PdClient:  pdClient,
+				},
+				Cluster:           cluster,
+				ClusterDeployment: clusterdeployment,
```

Comment:
agree i guess this will happen when we start to see more alert setups added, also it will be easier to see what we really need in terms of different setups